xlators/storage/posix/src/posix-common.c

```diff
@@ -1231,6 +1266,111 @@ posix_init(xlator_t *this)
+                    if (ret != -1) {
+                        /* link_path will be like this
+                           ../../<fhash>/<shash>/gfid_str so to populate
+                           absolute path from link path move the base_ptr to 6
+                           position and form an abosolute path and then read
+                           gfid from a absolute path and compare with
+                           rename_gfid
+                        */
+                        snprintf(abs_path, sizeof(abs_path), "%s/%s",
+                                 dir_handle, (link_path + 6));
+                        size = sys_lgetxattr(abs_path, "trusted.gfid", abs_gfid,
+                                             16);
+                        if (size == 16) {
+                            if (!gf_uuid_compare(abs_gfid, rename_gfid)) {
+                                posix_handle_unset(this, rename_gfid, NULL);
+                                unlink_flag = 0;
+                                ret = sys_rename(rename_gfid_path, gfid_path);
+                                if (ret) {
+                                    gf_log(this->name, GF_LOG_ERROR,
+                                           "rename gfid %s to expected gfid %s "
+                                           "is failed",
+                                           rename_gfid_path, gfid_path);
+                                } else {
+                                    gf_log(this->name, GF_LOG_INFO,
+                                           "rename gfid %s to expected gfid %s "
+                                           "is failed",
+                                           rename_gfid_path, gfid_path);
+                                }
+                            }
+                        } else {
+                            gf_log(this->name, GF_LOG_INFO,
+                                   "rename was not happened during crash so "
+                                   "unlink %s path",
+                                   rename_gfid_path);
+                        }
+                    }
+                    if (unlink_flag)
+                        (void)sys_unlink(rename_gfid_path);
+                }
+                ret = sys_unlinkat(_private->arrdfd[i], dentry->d_name);
+                if (ret) {
+                    gf_log(this->name, GF_LOG_INFO,
+                           "Unlink entry %s is failing in"
+                           " directory %s",
+                           dentry->d_name, dir_name);
+                }
+            }
+            sys_closedir(drptr);
+        }
+    }
+
+    xattr_value[0] = '0';
+    /*Poplate or Reset trusted.graceful_reboot xattr */
+    ret = sys_lsetxattr(_private->base_path, "trusted.graceful_reboot",
+                        xattr_value, 1, 0);
+    if (ret) {
+        gf_log(this->name, GF_LOG_INFO,
+               "Reset graceful_reboot xattr"
+               " value is failed");
+    }
```

Comment:
If we depend on this to decide whether to check for unclean shutdown, we cannot allow starting the brick if we can't set this value.

Reply:
Remove the code, it is not required